Fix Pyright errors in utils/types.py

XueSongTap opened the pull request:
fix #30147

akuporos commented on src/bindings/python/src/openvino/utils/types.py:
```diff
@@ -161,15 +161,15 @@ def make_constant_node(value: NumericData, dtype: Union[NumericType, Type] = Non
 
 
-def as_nodes(*input_values: NodeInput, name: Optional[str] = None) -> List[Node]:
+def as_nodes(*input_values: NodeInput, name: Optional[str] = None) -> List[Union[Node, Output]]:
```
```suggestion
def as_nodes(*input_values: NodeInput, name: Optional[str] = None) -> List[Union[Node, Output, Constant]]:
```
because `as_node` returns `Union[Node, Output, Constant]` several lines above

mlukasze commented:
linters check failed, fix it, please

p-wysocki commented:
Thank you for the PR, the fix looks great! Could you please also regenerate `.pyi` files as mentioned in the issue description? All you need to do is build OpenVINO with `-DENABLE_WHEEL=ON`, install that wheel `python -m pip install build/wheels/openvino...` and either:

- make any commit, and the `.pyi` files will be automatically regenerated and added to your commit
- run `python src/bindings/python/scripts/generate_stub_files.py` and `git add` them manually

akuporos commented:
Hello @XueSongTap,

There is still no updated .pyi file.
Could you please add it and I'll approve your PR :)

XueSongTap commented:
@a-noskov Sorry for the wrong commit, I've fixed the linters and added .pyi file. 

I would appreciate if you could review these changes at your convenience. Thank you for your patience.

```bash
(ov-env) yxc@yxc-MS-7B89:~/code/2504/openvino/src/bindings/python$ python -m pyright ./src/openvino/utils/types.py -p ./pyproject.toml --verbose
Loading pyproject.toml file at /home/yxc/code/2504/openvino/src/bindings/python/pyproject.toml
Assuming Python version 3.12.3.final.0
Execution environment: python
  Extra paths:
    /home/yxc/code/2504/openvino/src/bindings/python/src
  Python version: 3.12.3.final.0
  Python platform: Linux
  Search paths:
    /home/yxc/code/2504/ov-env/lib/python3.12/site-packages/pyright/dist/dist/typeshed-fallback/stdlib
    /home/yxc/code/2504/openvino/src/bindings/python
    /home/yxc/code/2504/openvino/src/bindings/python/src
    /home/yxc/code/2504/openvino/src/bindings/python/typings
    /home/yxc/code/2504/ov-env/lib/python3.12/site-packages/pyright/dist/dist/typeshed-fallback/stubs/...
    /home/yxc/code/2504/openvino/bin/intel64/Release/python
    /home/yxc/code/2504/openvino/tools/ovc
    /home/yxc/code/2504/openvino/src/bindings/python
    /usr/lib/python3.12
    /usr/lib/python3.12/lib-dynload
    /home/yxc/code/2504/ov-env/lib/python3.12/site-packages
Found 1 source file
pyright 1.1.399
0 errors, 0 warnings, 0 informations
Completed in 0.715sec
```

mlukasze commented:
linter failed again

p-wysocki commented:
Thank you for the investigation! The tasks for fixing `pyright` seemed straightforward, but when reviewing the PRs we decided that it's a much more difficult issue than we expected and, combined with the task priority, we decided to move fixes for `pyright` checks back as an internal OpenVINO team effort. It's no longer an open Good First Issue - we apologize for closing the task you were working on, your contribution helped us investigate that issue.
